### PR TITLE
Adding DAGMan guides

### PR DESCRIPTION
**Pull request by aowen-uwmad**
Updated the "Submit Workflow's with HTCondor's DAGMan" guide to include an introduction and to add ~quick-reference for using DAGMan.
Created new guide "Simple Example of a DAGMan Workflow". Gives step by step walkthrough of creating a simple A-->B DAG workflow and submitting it using DAGMan.

### CODE REVIEW

File: documentation/htc_workloads/automated_workflows/dagman-workflows.md [hunk 1]
```diff
@@ -3,28 +3,307 @@ ospool:
...
+```
+
+As a general rule, you should not set the number of retry attempts to more than 1 or 2 times. 
+If a job is failing repeatedly, it is better to troubleshoot the cause of that failure.
+This is especially true when you applying the `RETRY` statement to all of the nodes in your DAG.
+
+DAGMan considers the exit code of the last executed step when it considers the success or failure of the node overall.
+There are various possible combinations that can determine the success or failure of the node itself, as discussed in the HTCondor documentation [here](https://htcondor.readthedocs.io/en/latest/automated-workflows/node-pass-or-fail.html#).
+DAGMan only considers the success/failure of the node as a whole when deciding if it needs to attempt a retry.
+Importantly, if the `.sub` file for a node submits multiple HTCondor jobs, when any one of those jobs fails, DAGMan considers all of the jobs to have failed and will remove them from queue.
```
Comment (ChristinaLK) [inline documentation/htc_workloads/automated_workflows/dagman-workflows.md:246]:
```suggestion
**Importantly, if the `.sub` file for a node submits multiple HTCondor jobs, when any one of those jobs fails, DAGMan considers all of the jobs to have failed and will remove them from queue.**
```

File: documentation/htc_workloads/automated_workflows/dagman-workflows.md [hunk 2]
```diff
@@ -3,28 +3,307 @@ ospool:
...
+In the [DAGMan Features](#dagman-features) section, we will discuss tools that can assist you with this endeavor.
+
+## Running a DAG Workflow
+
+### 1. Submitting the DAG
+
+Because the DAG workflow represents a special type of job, a special command is used to submit it. To submit the DAG workflow, use 
+
+```
+condor_submit_dag input.dag
```
Comment (ChristinaLK) [inline documentation/htc_workloads/automated_workflows/dagman-workflows.md:105]: maybe use something other than `input`? use something like `run` or `workflow`

**Review by ChristinaLK — APPROVED** [review]
@aowen-uwmad I think this is fine to merge as is, and then edit in a new, separate PR if you really just want to get it on the website. Or you can address changes in this one. Up to you.

**Comment by aowen-uwmad** [comment]
Will make changes after merging.

**Comment by aowen-uwmad** [comment]
Changes incorporated in https://github.com/osg-htc/user-documentation/commit/910704eb00492bca8e81c5c4cd4aa21b4cc2960c.

**Pull request merged**